src/functionmap.jl

```diff
@@ -30,13 +30,31 @@ function Base.show(io::IO, A::FunctionMap{T}) where {T}
 end
 
 # properties
+Base.size(A::FunctionMap, n) = n==1 ? A.M : n==2 ? A.N : error("LinearMap objects have only 2 dimensions")
 Base.size(A::FunctionMap) = (A.M, A.N)
 LinearAlgebra.issymmetric(A::FunctionMap) = A._issymmetric
 LinearAlgebra.ishermitian(A::FunctionMap) = A._ishermitian
 LinearAlgebra.isposdef(A::FunctionMap)    = A._isposdef
 ismutating(A::FunctionMap) = A._ismutating
 _ismutating(f) = first(methods(f)).nargs == 3
 
+# special transposition behavior
+function LinearAlgebra.transpose(A::FunctionMap)
+    if A.fc!==nothing || A._issymmetric
```

Comment:
This should probably check that the linear map is real. We could also choose to make `transpose` and `adjoint` of a `FunctionMap` return a new function map, though this might not be type stable as it depends on `A._ishermitian`

Reply:
I'm not sure about the real check: we don't have one in the `At_mul_B!`, but use `f` for the transpose if `A` is symmetric in any case.

Regarding the new `FunctionMap` idea, as already mentioned I did experiment with that, but then I had two issues, IIRC. First, type stability as you say, and second I had difficulties of checking correctly for hermitianness of `A'A`, where `A::FunctionMap`.

Reply:
Yes, but if the user specified that `A` is symmetric, then `f` can be used for both `A_mul_B` and `At_mul_B`, even if the action is complex-valued. But if it is not symmetric, then you only want to use `fc` for `At_mul_B` if the linear map is real, otherwise `fc` is supposed to be the adjoint, and thus different from the transpose.

Hence, the test (if it stays) should probably be:
`A._issymmetric || (eltype(A) <: Real && A.fc !== nothing)`

Reply:
You're right. I messed it up with all the combinations.